model/histogram/generic.go

```diff
@@ -393,7 +406,44 @@ func checkHistogramBuckets[BC BucketCount, IBC InternalBucketCount](buckets []IB
 	return nil
 }
 
-func getBound(idx, schema int32) float64 {
+func checkHistogramCustomBounds(bounds []float64, spans []Span) error {
```

Comment:
Let's merge the functionality with `checkHistogramSpans` to avoid iterating the spans twice.